### PR TITLE
Support annotations on services created by controller

### PR DESCRIPTION
**Pull request by Matthew-Beckett**
This PR adds support to pass annotations in via the CRD spec which will be appended to the service created, this allows the usage of custom annotations which is required for configuring LoadBalancer implementations.

### CODE REVIEW

**Comment by Matthew-Beckett** [comment]
Depends on #35

**Comment by Matthew-Beckett** [comment]
@jodevsa How do I re-generate the CRD interaface? `make generate`?

**Comment by jodevsa** [comment]
Hi @Matthew-Beckett,

Please also try `make manifests`.


Btw, I have created a github job to create a release from a repo/branch. This is useful if you wanna easily try your changes on your cluster. 

Steps to use:

1- head to the actions page and click on the "DEV release" action
Screenshot 2023-04-29 at 16 25 23

2- You should see a button on the right that is called "Run workflow"

3- fill the form with your repo/branch


4- Trigger the workflow by pressing "Run workflow"


5- wait for the build to finish

6- Once your job is done, press on it

7- You should see an "artifacts" section with a release.yaml file
8- click on the file to download it
9- kubectl apply -f ~/fileLocation/release.yaml



let me know if you have any questions :)

**Comment by Matthew-Beckett** [comment]
@jodevsa I am not a contributor/maintainer so I do not have permission to run manual actions. May I recommend migrating this to a GitHub Organisation?

**Comment by jodevsa** [comment]
> @jodevsa I am not a contributor/maintainer so I do not have permission to run manual actions. May I recommend migrating this to a GitHub Organisation?

Sure, let me have a look

**Comment by jodevsa** [comment]
@Matthew-Beckett Can you try again? I gave permissions to everyone to run that flow. I'll have a look at moving this to an organisation soon

**Comment by jodevsa** [comment]
I've also added you as a collaborator. please accept the invite.

**Comment by Matthew-Beckett** [comment]
> I've also added you as a collaborator. please accept the invite.

Done, thank you!

File: go.mod
Comment (jodevsa) [inline go.mod:3]: Do you need go.sum and go.mod to be updated? if not, I would prefer to postpone updating any dependencies in a separate PR if possible.

**Review by jodevsa — APPROVED** [review]
LGTM, it would be nice if you can remove the go.mod/go.sum changes and do that in a separate PR if possible

**Comment by jodevsa** [comment]
btw, did the release workflow work for you? I also wonder if the arm images are working fine

**Comment by Matthew-Beckett** [comment]
@jodevsa the dependency update for wgctl is to fix a build failure due to missing artefact, I am not sure how the other changes appeared.

Again, not a great Go dev, could you help me understand where they came from and how to revert them?

**Pull request merged**
